### PR TITLE
Visual improvements to DataTable

### PR DESCRIPTION
**Pull request by maxiadlovskii**
## Description
This PR adds a few new props for more visual flexibility in DataTable component

-   striped
-   bordered
-   stickyHeader
-   borderedHeader

Also, Pivot Fields Headers cells are now centered. We need that because data in the table and the columns headers might be aligned depending on the value type ( left - string, right - number). In this case for the user is hard to understand to which precisely the columns pivot header relate. The situation became more worster when the user has several columns for one pivot header. 
Screenshot 2022-09-07 at 12 28 59
With a centered pivot header and borderedHeader for the user would be easier to look at the data   
Screenshot 2022-09-07 at 12 32 34

## Motivation and Context
#11675
https://github.com/Graylog2/graylog-plugin-enterprise/issues/3499
[#10936](https://github.com/Graylog2/graylog2-server/issues/10936)
[#11675](https://github.com/Graylog2/graylog2-server/issues/11675)
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/webpack/vendor-module-ids.json [hunk 1]
```diff
@@ -542,6 +542,8 @@
       "./node_modules/value-equal/esm/value-equal.js": 5429,
       "./node_modules/warning/browser.js": 670,
       "dll vendor": 338,
+      "javascript/dynamic|./node_modules/@babel/runtime/helpers/extends.js": 434,
+      "javascript/dynamic|./node_modules/@babel/runtime/helpers/interopRequireDefault.js": 836,
```
Comment (kmerz) [inline graylog2-web-interface/webpack/vendor-module-ids.json:546]: I might be wrong but are these changes really necessary?

**Review by kmerz — COMMENTED** [review]
I could not tested yet, but I left some questions I would need an answer for, since I am not so deep in the code anymore.
I will test tomorrow in the morning.

File: graylog2-web-interface/src/views/components/datatable/MessagesTable.tsx [hunk 2]
```diff
@@ -26,18 +26,28 @@ const MessagesContainer = styled.div`
...
+  
+  thead {
+  ${$stickyHeader
+    ? `position: sticky;
+    top: 0;
+    z-index: 1` : ''}
+  }
+  & th {
 
+  }
```
[the remark this answers is not in the document]
Comment (maxiadlovskii) [reply graylog2-web-interface/src/views/components/datatable/MessagesTable.tsx:45]: I forgot to remove it. Thanks

File: graylog2-web-interface/src/views/components/datatable/DataTableEntry.tsx [hunk 3]
```diff
@@ -117,20 +117,18 @@ const DataTableEntry = ({ columnPivots, currentView, fields, series, columnPivot
   const columns = flatten([fieldColumns, columnPivotFields]);
 
   return (
-    <tbody className={classes}>
```
[the remark this answers is not in the document]
Comment (maxiadlovskii) [reply graylog2-web-interface/src/views/components/datatable/DataTableEntry.tsx:120]: I moved that class to the next tag 
` <tr className={`fields-row ${classes}`}>`
`message-group ` removing top border 
In the next commit, I will change 
```
  tbody.message-group {
    border-top: 0;
  }
```
   with 
```
     tr.message-group {
    border-top: 0;
  }
```

File: graylog2-web-interface/webpack/vendor-module-ids.json (hunk 1 above)
Comment (maxiadlovskii) [reply graylog2-web-interface/webpack/vendor-module-ids.json:546]: Looks like it was changed automatically... I don't know why exactly. I will remove it and check

File: graylog2-web-interface/src/views/components/datatable/Headers.tsx [hunk 4]
```diff
@@ -32,6 +32,10 @@ const StyledTh = styled.th(({ isNumeric }: { isNumeric: boolean }) => css`
   ${isNumeric ? 'text-align: right' : ''}
 `);
 
+const CentratedTh = styled.th`
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/datatable/Headers.tsx:35]: Suggestion for better naming:

```suggestion
const CenteredTh = styled.th`
```

File: graylog2-web-interface/src/views/components/datatable/Headers.tsx [hunk 5]
```diff
@@ -43,17 +47,19 @@ const _headerField = (activeQuery: string, fields, field: string, prefix: (strin
...
+  </CentratedTh>
 );
 
 // eslint-disable-next-line jsx-a11y/control-has-associated-label
 const _spacer = (idx, span = 1) => <th colSpan={span} key={`spacer-${idx}`} className={styles.leftAligned} />;
 
 const columnPivotFieldsHeaders = (activeQuery: string, columnPivots: string[], actualColumnPivotValues: any[], series: Series[], offset = 1) => {
   return columnPivots.map((columnPivot, idx) => {
-    const actualValues = actualColumnPivotValues.map((key) => ({ path: key.slice(0, idx).join('-'), key: key[idx] || '', count: 1 }));
+    const actualValues = actualColumnPivotValues.map((key) => {
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/datatable/Headers.tsx:60]: I think this change can be removed. Using an arrow function without curly braces is perfectly fine.

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, thank you! Just two minor suggestions inline and please remove the changes to `graylog2-web-interface/webpack/vendor-module-ids.json`.

**Pull request merged**